### PR TITLE
Update rules_go

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

Updates rules go to the latest release. Includes upgrade to go 1.15.1.

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**

### CODE REVIEW

**Review by terencechain — APPROVED** [review]

**Comment by prestonvanloon** [comment]
This breaks mac OS X cross compile, but doesn't break native mac os x builds. 

There is no clear error or indication why it won't build.

**Comment by prestonvanloon** [comment]
Update on this: go 1.15 no longer supports Mac OS X lower than 10.14. 
We were using the Mac OS X SDK from 10.10 to build cross compile. 
The SDK will need to be updated to 10.14 or greater. We are working through this now.

**Comment by protolambda** [comment]
Something in discv5 was broken in Go 1.15. It still works running against other Go implementations (when they are also on Go 1.15), but we are hitting interop issues against others (experienced this with my bootnode and rumor programs). Felix fixed made a fix here: https://github.com/ethereum/go-ethereum/pull/21673

Go 1.15 <> 1.14 discv5, or any other discv5 implementation, won't work without this fix.

**Comment by prestonvanloon** [comment]
Thanks @protolambda! We'll hold off until that PR merges.

**Comment by fjl** [comment]
It's merged to go-ethereum master now

**Comment by rkapka** [comment]
I added some `-I` flags to include a folder, but it didn't help :(

**Comment by protolambda** [comment]
Can this PR reopen? Updating to new Go version would be great. Discv5 issues with 1.15 have been resolved a while ago (just update the geth dependency if you have not already).

**Comment by prestonvanloon** [comment]
I spent another 4 hours on this and we're still stuck.

This is not a priority and may not happen for some time.

**Comment by prestonvanloon** [comment]
Getting closer to having this ready to merge. Still need to fix libfuzzer builds, and something with cgo BLS.

File: tools/cross-toolchain/regenerate.sh
Comment (farazdagi) [inline tools/cross-toolchain/regenerate.sh:12]: Do we need to keep this commented?

Comment (prestonvanloon) [reply tools/cross-toolchain/regenerate.sh:12]: No, we are using clang only at the moment.

Comment (prestonvanloon) [reply tools/cross-toolchain/regenerate.sh:12]: I'll remove it

**Review by rauljordan — APPROVED** [review]

**Review by farazdagi — APPROVED** [review]

**Pull request merged**
